sync/clients/__init__.py

```diff
@@ -15,3 +17,54 @@ def encode_json(obj: dict) -> tuple[dict, str]:
         "Content-Length": str(len(json)),
         "Content-Type": "application/json",
     }, json
+
+
+class RetryableHTTPClient:
+    """
+    Smaller wrapper around httpx.Client/AsyncClient to contain retrying logic that httpx does not offer natively
+    """
+
+    _DEFAULT_RETRYABLE_STATUS_CODES: set[httpx.codes] = {
+        httpx.codes.REQUEST_TIMEOUT,
+        httpx.codes.TOO_EARLY,
+        httpx.codes.TOO_MANY_REQUESTS,
+        httpx.codes.INTERNAL_SERVER_ERROR,
+        httpx.codes.BAD_GATEWAY,
+        httpx.codes.SERVICE_UNAVAILABLE,
+        httpx.codes.GATEWAY_TIMEOUT,
+    }
+
+    def __init__(self, client: httpx.Client | httpx.AsyncClient):
+        self._client: httpx.Client | httpx.AsyncClient = client
+
+    def _send_request(self, request: httpx.Request) -> httpx.Response:
+        try:
+            for attempt in Retrying(
+                stop=stop_after_attempt(3),
+                wait=wait_exponential_jitter(initial=2, max=10, jitter=2),
+            ):
+                with attempt:
+                    response = self._client.send(request)
+                    if response.status_code in self._DEFAULT_RETRYABLE_STATUS_CODES:
+                        raise TryAgain
```

Comment:
`TryAgain` seems to do the trick as well :+1:

Reply:
Do you think it's a typo in their documentation to not instantiate `TryAgain`? I don't think I've seen that before. Oh, maybe just not recommended: https://peps.python.org/pep-0317/

Reply:
Ah, yeah, I was curious about that... it did work when I tried, though!

Reply:
Works for me too. Interesting though that after the DNS change I get a 502 back from staging instead of a 503...